### ISSUE TITLE
Sign of kWh usage in TariffTransaction

### ISSUE THREAD

**Issue by pascalwhoop** [powertac/powertac-server#966]
The spec PDF says

> Quantities of money and energy in TariffSpecifications and associated structures are represented
> from the viewpoint of a Customer. [...] Similarly, a positive quantity of energy represents energy 
> delivered to the Customer, and a negative quantity represents energy delivered to the Broker. 

The code says

```
    0   /** The total kWh of energy consumed (< 0) or produced (> 0) in kWh.                                                                                                                      
    1    *  Note that this is not per-individual in a population model, but rather                                                                                                                
    2    *  aggregate usage by customerCount individuals. */                                                                                                                                      
    3   @XStreamAsAttribute                                                                                                                                                                       
    4   private double kWh = 0.0;      
```

which is true? My analysis of aggregate use over all customers suggests the code is right. So a bug in the spec document? 

Kind regards,
Pascal

**Comment by jecollins** [comment]
The confusion is not about the signs, but about what are "associated structures". That was meant to refer to the Rates, and should be clarified. The more general principle is that messages are signed from the viewpoint of the message's intended recipient. In the case of TariffSpecification, the intended recipient is a Customer; for TariffTransaction (where the code documentation you cite comes from), the intended recipient is the Broker.

**Comment by pascalwhoop** [comment]
OK that’s good to know. I was able to figure it out because I looked at the data. But I could see someone else being confused unless the view point situation for the intended recipient that you described is also documented. Anyways thanks for the clarification. I saw that this year‘s finals dates are now online so I will have to make sure my time schedule aligns with the dates. Right now I am still mostly doing preprocessing and scaffolding my agent in python so that I can easily parse both XML and state files.

Overall I really like the topic and the entire simulation a lot of though and I keep catching myself working on this for hours even late at night. So thanks for the project.

Kind regards,
Pascal
